src/app/services/text_generation.py

```diff
@@ -40,7 +40,8 @@ class TextGenerationConfig(ContainerConfig):
     quantize: Optional[str] = None
     revision: Optional[str] = None
     validation_workers: Optional[int] = None
-    sharded: Optional[bool] = True
+    sharded: Optional[str] = None  # Need to set up as 'true' or 'false'
```

Comment:
What's this comment mean?

Reply:
The HuggingFace API only accepts shared as the string ("true" of "false'). Passing True or False to the HF API would lead to errors.

Reply:
This is an optional argument in TGI API launcher, so it's optional here as well. The template just leaves it out if it's `None`:
```python
{%- if 'sharded' in container_config %}
  --sharded {{ container_config['sharded'] }} \
{%- endif %}
```

Reply:
sharded: Optional[Literal["true", "false"]] = None. This is good!